lib/ui/src/modules/ui/libs/hierarchy.js

```diff
@@ -46,24 +46,56 @@ function fillHierarchy(namespaces, hierarchy, story) {
   fillHierarchy(namespaces.slice(1), childHierarchy, story);
 }
 
-export function createHierarchy(stories) {
-  const hierarchyRoot = {
+export function createHierarchyRoot(name = '') {
+  return {
     isNamespace: true,
     namespaces: [],
-    name: '',
+    name,
     map: new Map(),
   };
+}
+
+export function createHierarchies(stories) {
+  const rootMap = {};
 
   if (stories) {
     stories.forEach(story => {
-      const { namespaces } = story;
+      const { rootName = '', namespaces } = story;
       const name = namespaces[namespaces.length - 1];
+      const hierarchyRoot =
+        rootMap[rootName] || (rootMap[rootName] = createHierarchyRoot(rootName));
 
       fillHierarchy(namespaces, hierarchyRoot, { ...story, name });
     });
   }
 
-  return hierarchyRoot;
+  const rootNames = Object.keys(rootMap);
+  const hierarchies = rootNames.map(rootName => rootMap[rootName]);
+
+  // make sure the main root is the first
```

Comment:
Please extract this block to function - `moveRootToFront` / `ensureRootIsFirst` or something

Reply:
Extracted to `ensureMainRootIsFirst` and used `Object.values` instead of `Object.keys(...).map(...)`